src/aleph/web/controllers/p2p.py

```diff
@@ -11,21 +16,44 @@
 LOGGER = logging.getLogger("web.controllers.p2p")
 
 
-async def pub_json(request):
+def validate_request_data(config: Config, request_data: Dict) -> None:
+    """
+    Validates the content of a JSON pubsub message depending on the channel
+    and raises a 422 error if the data does not match the expected format.
+
+    :param config: Application configuration.
+    :param request_data: Request JSON data, as a dictionary.
+    """
+
+    topic = request_data.get("topic")
+    data = request_data.get("data")
+
+    # Currently, we only check validate messages
+    message_topic = config.aleph.queue_topic.value
+    if topic == message_topic:
+        try:
+            _ = parse_message(data)
+        except InvalidMessageError as e:
+            raise web.HTTPUnprocessableEntity(text=str(e))
```

Comment:
The error message should, when the issue was raised by Pydantic, use the content from ValidationError.json.

Reply:
Okay sure, fine by me.